src/main.cc

```diff
@@ -1320,223 +917,227 @@ gint main(gint argc, gchar *argv[])
 
-			work = command_line->cmd_list;
-			while (work)
-				{
-				FileData *fd;
+	g_application_hold(G_APPLICATION(app));
+}
 
-				fd = file_data_new_simple(static_cast<const gchar *>(work->data));
-				collection_add(cd, fd, FALSE);
-				file_data_unref(fd);
-				work = work->next;
-				}
+} // namespace
 
-			work = command_line->collection_list;
-			while (work)
-				{
-				collection_load(cd, static_cast<gchar *>(work->data), COLLECTION_LOAD_APPEND);
-				work = work->next;
-				}
+void exit_program()
+{
+	layout_image_full_screen_stop(nullptr);
 
-			if (cd->list) layout_image_set_collection(nullptr, cd, static_cast<CollectInfo *>(cd->list->data));
+	if (metadata_write_queue_confirm(FALSE, exit_program_write_metadata_cb, nullptr)) return;
 
-			/* mem leak, we never unref this collection when !startup_command_line_collection
-			 * (the image view of the main window does not hold a ref to the collection)
-			 * this is sort of unavoidable, for if it did hold a ref, next/back
-			 * may not work as expected when closing collection windows.
-			 *
-			 * collection_unref(cd);
-			 */
+	options->marks_save ? marks_save(TRUE) : marks_save(FALSE);
 
-			}
-		else if (first_collection)
-			{
-			layout_image_set_collection(nullptr, first_collection,
-						    collection_get_first(first_collection));
-			}
+	if (exit_confirm_dlg())
+		{
+		return;
+		}
+
+	exit_program_final();
+}
 
-		/* If the files on the command line are from one folder, select those files
-		 * unless it is a command line collection - then leave focus on collection window
-		 */
-		lw = nullptr;
-		layout_valid(&lw);
+gint main(gint argc, gchar *argv[])
+{
+	gdk_set_allowed_backends("x11, *");
+
+	gint status;
+	gchar *version_string;
+	GtkApplication *app;
+	// We handle unit tests here because it takes the place of running the
+	// rest of the app.
+	if (search_command_line_for_unit_test_option(argc, argv))
+		{
+#if ENABLE_UNIT_TESTS
+		testing::InitGoogleTest(&argc, argv);
+		return RUN_ALL_TESTS();
+#else
+		fprintf(stderr, "Unit tests are not enabled in this build.\n");
+		return 1;
+#endif
+		}
 
-		if (single_dir && command_line->cmd_list && !command_line->startup_command_line_collection)
+#if HAVE_CLUTTER
+	if (! g_getenv("GQ_DISABLE_CLUTTER"))
+		{
+		if (clutter_init(nullptr, nullptr) != CLUTTER_INIT_SUCCESS)
 			{
-			GList *work;
-			GList *selected;
-			FileData *fd;
+			fprintf(stderr, \
+_("Can't initialize clutter-gtk. \n \
+To start Geeqie use: \n \
+GQ_DISABLE_CLUTTER=1 geeqie\n\n"));
 
-			selected = nullptr;
-			work = command_line->cmd_list;
-			while (work)
-				{
-				fd = file_data_new_simple(static_cast<gchar *>(work->data));
-				selected = g_list_append(selected, fd);
-				file_data_unref(fd);
-				work = work->next;
-				}
-			layout_select_list(lw, selected);
+			return EXIT_FAILURE;
 			}
+		}
+#endif
+
+	if (g_getenv("GQ_CACHE_MAINTENANCE"))
+		{
+		/* Disabled at the moment */
+		log_printf("Command line cache maintenance is disabled");
```

Comment:
If this is going to be submitted (even temporarily), it might be good to add "...is disabled until the code can be fixed" or something.  I can imagine a user seeing this warning as it's currently written and wondering what they're doing wrong to enable it.